staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -87,6 +87,7 @@ type Controller struct {
 	eventRecorder       record.EventRecorder
 	nodeLister          corelisters.NodeLister
 	nodeListerSynced    cache.InformerSynced
+	endpointsLister     corelisters.EndpointsLister
```

Comment:
New code should not be using Endpoints, since it doesn't support dual-stack, topology, or various other features. You need to use EndpointSlices.